### PR TITLE
Remove Python 2 from bottle

### PR DESCRIPTION
**Pull request by xmonader**
- What I did is basically removed any py3k usages and removed all of the else branches that covers 2.7
- Updated the imports in the rest of the codebase
- Removed Python 2.7 in the code/doc/test base
- Tests are Green
- Code is black formatted

### CODE REVIEW

**Comment by oz123** [comment]
This is so much welcomed!

File: bottle.py [hunk 1]
```diff
@@ -874,26 +725,21 @@ def get_url(self, routename, **kargs):
...
-              callback=None,
-              name=None,
-              apply=None,
-              skip=None, **config):
+    def route(self, path=None, method='GET', callback=None, name=None,
+              apply=None, skip=None, **config):
         """ A decorator to bind a function to a request URL. Example::
 
-                @app.route('/hello/<name>')
+                @app.route('/hello/:name')
```
Comment (agalera) [inline bottle.py]: Is it compatible with the previous patterns?

**Comment by xmonader** [comment]
- All tests are green.
- Black formatted.

File: bottle.py [hunk 2]
```diff
@@ -2,116 +2,52 @@
...
-        from inspect import getfullargspec
-        def getargspec(func):
-            spec = getfullargspec(func)
-            kwargs = makelist(spec[0]) + makelist(spec.kwonlyargs)
-            return kwargs, spec[1], spec[2], spec[3]
-    except ImportError:
-        from inspect import getargspec
-
-py3k = sys.version_info.major > 2
+from json import dumps as json_dumps, loads as json_lds
```
[the remark this answers is not in the document]
Comment (oz123) [reply bottle.py]: ujson iis pretty much abandoned

**Comment by defnull** [comment]
Sorry, but this PR is way to big to be merged as one giant patch set. I cannot possibly review 2600+ changes in a single patch and be sure that there are no breaking changes. Large patch sets like these would also destroy the git history (git blame) and break all other PRs.

Please, if we do this, then step by step in a manageable way.

* Fix one part/function/class at a time. Small trivial patches are way easier to review.
* Do not mix functional and formatting changes. Do not touch more lines than absolutely necessary.
* For any non-trivial fix, commit code and test changes separately. Fix the code first, and see if the tests still pass. We need a way to verify that the behavior did not change and that we did not break anything.
* The Python 2.7 EOL date is 1.1.2020. That's less than three months. See #1162 for why we won't be able to merge any breaking changes before that date.

And please, DO NOT run automated tools (2to3, black) over the entire source tree. I cannot stress that enough. These tools change way too much in one step. The patches are unreadable, destroy the git history (git blame) and break all other merge requests that might still be relevant. I will not merge patches created by these tools.

I'll close this PR, but you can continue discussing in #1162 if you want.

**Pull request closed without merging**